src/Tizen.MachineLearning.Train/Tizen.MachineLearning.Train/Model.cs

```diff
@@ -0,0 +1,409 @@
+            if (disposing)
+            {
+                // release managed object
+            }
+            // release unmanaged object
+            if (handle != IntPtr.Zero)
+            {
+                // Destroy the neural network model.
+                NNTrainerError ret = Interop.Model.Destroy(handle);
+                NNTrainer.CheckException(ret, "Failed to destroy model instance");
+
+                handle = IntPtr.Zero;
+            }
+            disposed = true;
+        }
+
+        /// <summary>
+        /// Compiles and finalizes the neural network model with the hyperparameter.
+        /// </summary>
+        /// <remarks>
+        /// Use this function to initialize neural network model.Various
+        /// hyperparameter before compile the model can be set. Once compiled,
+        /// any modification to the properties of model or layers/dataset/optimizer in
+        /// the model will be restricted. Further, addition of layers or changing the
+        /// optimizer/dataset of the model will not be permitted.
+        /// The input format of hyperparameter must be 'key = value' format.
+        /// <param name="hyperparameter">Hyperparameters for train complie.</param>
+        /// </remarks>
+        /// <since_tizen> 10 </since_tizen>
+        public void Compile(params string[] hyperparameter)
+        {
+            string compileParams = null;
+
+            if (hyperparameter.Length > 0) {
+                compileParams = string.Join("|", hyperparameter);
+                Log.Info(NNTrainer.Tag, "Compile hyperparameter:"+ compileParams);
+            }
+
+            NNTrainerError ret = Interop.Model.Compile(handle, compileParams);
+            NNTrainer.CheckException(ret, "Failed to compile model");
+        }
+
+        /// <summary>
+        /// Trains the neural network model with the hyperparameter.
+        /// </summary>
+        /// <remarks>
+        /// Use this function to train the compiled neural network model with
+        /// the passed training hyperparameters. This function will return once the
+        /// training, along with requested validation and testing, is completed.
+        /// The input format of hyperparameter must be 'key = value' format.
+        /// </remarks>
+        /// <param name="hyperparameter">Hyperparameters for train model.</param>
+        /// <since_tizen> 10 </since_tizen>
+        public void Run(params string[] hyperparameter)
+        {
+            string runParams = null;
+
+            if (hyperparameter.Length > 0) {
+                runParams = string.Join("|", hyperparameter);
+                Log.Info(NNTrainer.Tag, "Run hyperparameter:"+ runParams);
+            }
+
+            NNTrainerError ret = Interop.Model.Run(handle, runParams);
+            NNTrainer.CheckException(ret, "Failed to run model");
+        }
+
+        /// <summary>
+        /// Gets the summary of the neural network model.
+        /// </summary>
+        /// <remarks>
```

Comment:
this remark does not add anything, could be dropped

Reply:
okay